src/wxflow/hsi.py

```diff
@@ -0,0 +1,442 @@
+                Whether the output of the hsi command should be written to stdout
+
+        ignore_errors : list
+                List of error numbers to ignore.  For example, hsi returns error
+                number 64 if a target file does not exist on HPSS.
+
+        Returns
+        -------
+        output : str
+                Concatenated output and error of the hsi command.
+
+        Example:
+        --------
+            >>> hsi = Hsi()
+            >>> # Execute `hsi get some_local_file : /some/hpss/file`
+            >>> hsi._hsi(["get","some_local_file : /some/hpss/file"])
+        """
+
+        # Remove any empty arguments which can cause issues for hsi
+        arg_list = [arg for arg in arg_list if arg != ""]
+
+        if silent:
+            output = self.exe(*arg_list, output=str, error=str,
+                              ignore_errors=ignore_errors)
+        else:
+            output = self.exe(*arg_list, output=str.split, error=str.split,
+                              ignore_errors=ignore_errors)
+
+        return output
+
+    def get(self, source: str, target=None, opts: Union[List, str] = []) -> str:
+        """ Method to get a file from HPSS via hsi
+
+        Parameters
+        ----------
+        source : str
+                Full path location on HPSS of the file
+
+        target : str
+                Location on the local machine to place the file.  If not specified,
+                then the file will be placed in the current directory.
+
+        opts : str | list
+                List or string of additional options to send to hsi command.
+
+        Returns
+        -------
+        output : str
+                Concatenated output and error of the hsi get command.
+        """
+        arg_list = []
+
+        # Convert to str to handle Path objects
+        source = str(source)
+        target = str(target) if target is not None else None
+
+        # Parse any hsi options
+        arg_list.extend(Hsi._split_opts(opts))
+
+        arg_list.append("get")
+        args_list.append(source) if target is None else arg_list.append(f"{target} : {source}")
+
+        output = self._hsi(arg_list)
+
+        return output
+
+    def put(self, source: str, target: str, opts: Union[List, str] = [],
+            listing_file: str = None) -> str:
+        """ Method to put a file onto HPSS via hsi
+
+        Parameters
+        ----------
+        source : str
+                Location on the local machine of the source file to send to HPSS.
+
+        target : str
+                Full path of the target location of the file on HPSS.
+
+        opts : str | List
+                List or string of additional options to send to hsi.
+
+        Returns
+        -------
+        output : str
+                Concatenated output and error of the hsi put command.
+        """
+        arg_list = []
+
+        # Convert to str to handle Path objects
+        target = str(target)
+        source = str(source)
```

Comment:
Is this necessary since the `type` is `str` as defined in the method signature?
If this could be a `Path`, then it makes sense.

Reply:
The pytests define these inputs as `Path`s.  The signature is a suggestion -- Python will accept whatever type you throw at it.